kronika.go

```diff
@@ -18,7 +18,8 @@ func Every(ctx context.Context, start time.Time, interval time.Duration) <-chan
 			total := diff - interval
 			times := total / interval * -1
 
-			start = start.Add(times * interval)
+			// see https://github.com/charithe/durationcheck
+			start = start.Add(time.Duration(times.Nanoseconds()*interval.Nanoseconds()) * time.Nanosecond)
```

Comment:
Some things to note:

1. `time.Nanosecond` is `1`
2. `duration.Nanoseconds()` simply converts the duration to `int64`

So what you've written is the equivalent of:

```go
start = start.Add(time.Duration(int64(times)*int64(interval)) * 1)
```

So you converted the duration to `int64`, multiplied them, and then converted it back to `time.Duration`.

If we remove the unnecessary type conversions, we go back to

```go
start = start.Add(times * interval * 1)
```

Which is identical to what is already there, and I think the current code is more readable.